src/scripts/main.js

```diff
@@ -1,3 +1,35 @@
 'use strict';
 
-// write code here
+const listOfItems = document.querySelectorAll('li');
+const originalList = document.querySelector('ul');
+
+function sortList(list) {
+  let newList = [...list];
+
+  newList = newList.sort((person1, person2) =>
+    convertToNumber(person2.dataset.salary)
+    - convertToNumber(person1.dataset.salary));
+
+  originalList.append(...newList);
+
+  getEmployees(newList);
+}
+
+function getEmployees(list) {
+  return list.map(person => {
+    const sortedPeople = {
+      name: person.innerText,
+      position: person.dataset.position,
+      salary: person.dataset.salary,
+      age: person.dataset.age,
+    };
```

Comment:
You can use destructuring for the `person.dataset` to improve the code readability
```
const { age, position, salary } = person.dataset;
```
```suggestion
    const sortedPeople = {
      name: person.innerText,
      position,
      salary,
      age,
    };
```